src/contracts/middleware/BLSSignatureChecker.sol

```diff
@@ -1,508 +1,218 @@
 // SPDX-License-Identifier: BUSL-1.1
 pragma solidity =0.8.12;
 
-import "../interfaces/IBLSRegistry.sol";
-import "../libraries/BytesLib.sol";
+import "../interfaces/IBLSRegistryCoordinatorWithIndices.sol";
 import "../libraries/MiddlewareUtils.sol";
 import "../libraries/BN254.sol";
+import "../libraries/BitmapUtils.sol";
 
 /**
  * @title Used for checking BLS aggregate signatures from the operators of a `BLSRegistry`.
  * @author Layr Labs, Inc.
  * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
  * @notice This is the contract for checking the validity of aggregate operator signatures.
  */
-abstract contract BLSSignatureChecker {
+contract BLSSignatureChecker {
+    using BN254 for BN254.G1Point;    
+
     // DATA STRUCTURES
-    /**
-     * @notice this data structure is used for recording the details on the total stake of the registered
-     * operators and those operators who are part of the quorum for a particular taskNumber
-     */
 
-    struct SignatoryTotals {
-        // total stake of the operators who are in the first quorum
-        uint256 signedStakeFirstQuorum;
-        // total stake of the operators who are in the second quorum
-        uint256 signedStakeSecondQuorum;
-        // total amount staked by all operators (irrespective of whether they are in the quorum or not)
-        uint256 totalStakeFirstQuorum;
-        // total amount staked by all operators (irrespective of whether they are in the quorum or not)
-        uint256 totalStakeSecondQuorum;
+    struct NonSignerStakesAndSignature {
+        uint32[] nonSignerQuorumBitmapIndices;
+        BN254.G1Point[] nonSignerPubkeys;
+        BN254.G1Point[] quorumApks;
+        BN254.G2Point apkG2;
+        BN254.G1Point sigma;
+        uint32[] quorumApkIndices;
+        uint32[] totalStakeIndices;  
+        uint32[][] nonSignerStakeIndices; // nonSignerStakeIndices[quorumNumberIndex][nonSignerIndex]
     }
```

Comment:
Can we document/define the purpose of this struct and all of these fields?
It's challenging to keep track of everything; especially things like `nonSignerQuorumBitmapIndices` don't necessarily immediately have clear meaning